src/models/sectionComponent.ts

```diff
@@ -24,4 +24,4 @@ export interface SectionProps {
  *
  * @public
  */
-export type SectionComponent = (props: SectionProps) => JSX.Element | null;
+export type SectionComponent<T> = (props: SectionProps<T>) => JSX.Element | null;
```

Comment:
if a user writes a custom section, will they be required to either specify a type or make it a generic?

Reply:
Updated to provide default typing!